tests/test_tenant_unauthorized.py

```diff
@@ -62,27 +449,77 @@ def mock_input_validation(mocker):
+    request,
+    mock_input_validation,
+    db,
+    userNoTenantPermissions,
+    groupNoTenantPermissions,
 ):
-    if _type.name == 'Mutation' and f'{_type.name}.{field.name}' in OPT_OUT_MUTATIONS.keys():
-        pytest.skip(f'Skipping test for {field.name}: {OPT_OUT_MUTATIONS[f"{_type.name}.{field.name}"]}')
-    if _type.name == 'Query' and f'{_type.name}.{field.name}' not in OPT_IN_QUERIES:
-        pytest.skip(f'Skipping test for {field.name}: This Query does not require a tenant permission check.')
+    fid = request.node.callspec.id
+    tdata = EXPECTED_RESOLVERS[fid]
+    msg = f'{fid} -> {field.resolver.__code__.co_filename}:{field.resolver.__code__.co_firstlineno}'
+    expected_perm = tdata.tenant_perm
+    if not expected_perm:
+        pytest.skip(msg + f' Reason: {tdata.tenant_ignore.value}')
+    logging.info(msg)
+
     assert_that(field.resolver).is_not_none()
-    mock_local.context = RequestContext(
-        db, userNoTenantPermissions.username, [groupNoTenantPermissions.groupUri], userNoTenantPermissions
-    )
+    groups = [groupNoTenantPermissions.groupUri]
+    username = userNoTenantPermissions.username
+    mock_local.context = RequestContext(db, username, groups, userNoTenantPermissions)
     # Mocking arguments
     iargs = {arg: MagicMock() for arg in inspect.signature(field.resolver).parameters.keys()}
     # Assert Unauthorized exception is raised
     assert_that(field.resolver).raises(TenantUnauthorized).when_called_with(**iargs).contains('UnauthorizedOperation')
+    spy_check.assert_called_once_with(
+        session=ANY, username=username, groups=groups, tenant_name=ANY, permission_name=expected_perm
+    )
+
+
+@patch('dataall.base.aws.sts.SessionHelper.remote_session')
+@patch('dataall.core.permissions.services.resource_policy_service.ResourcePolicyService.check_user_resource_permission')
+@patch('dataall.base.context._request_storage')
+@pytest.mark.parametrize('_type,field', ALL_PARAMS)
+def test_unauthorized_resource_permissions(
+    mock_local,
+    mock_check,
+    mock_session,
+    _type,
+    field,
+    request,
+):
+    fid = request.node.callspec.id
+    tdata = EXPECTED_RESOLVERS[fid]
+    msg = f'{fid} -> {field.resolver.__code__.co_filename}:{field.resolver.__code__.co_firstlineno}'
+    expected_perm = tdata.resource_perm
+    if not expected_perm:
+        pytest.skip(msg + f' Reason: {tdata.tenant_ignore.value}')
+    logging.info(msg)
+
+    assert_that(field.resolver).is_not_none()
+    username = 'ausername'
+    groups = ['agroup']
+    mock_local.context = RequestContext(MagicMock(), username, groups, 'auserid')
+    mock_local.context.db_engine.scoped_session().__enter__().query().filter().all.return_value = [MagicMock()]
+    mock_check.side_effect = ResourceUnauthorized(groups, 'test_action', 'test_uri')
+    iargs = {arg: MagicMock() for arg in inspect.signature(field.resolver).parameters.keys()}
+    with suppress(ResourceUnauthorized):
+        field.resolver(**iargs)
```

Comment:
Just to confirm my understanding. Here we are ignoring the exception because it is an artificial exception created by the mocker side effect in line 515. With this test we are assuming that the `ResourcePolicyService` returns `ResourceUnauthorized`, while for the tenantUnauthorized tests we explicitly checked that the resolver raised the expected exception

Reply:
Correct, the important part of both tests is that they call the corresponding check function with the specific user/group and permission, wether it throws or not is irrelevant. I also added this logic in the tenant test (check `spy_check`)
The reason I use the side_effect here is because I want to stop the execution as soon as we receive a call to the check to stop it from calling any of the business logic code which will produce a lot of errors etc.